src/codegen.cpp

```diff
@@ -285,14 +285,54 @@ static LLVMCallConv get_llvm_cc(CodeGen *g, CallingConvention cc) {
         case CallingConventionNaked:
             zig_unreachable();
         case CallingConventionStdcall:
-            // stdcall calling convention only works on x86.
-            if (g->zig_target->arch == ZigLLVM_x86) {
+            if (g->zig_target->arch == ZigLLVM_x86)
                 return LLVMX86StdcallCallConv;
-            } else {
-                return LLVMCCallConv;
-            }
+            return LLVMCCallConv;
+        case CallingConventionFastcall:
+            if (g->zig_target->arch == ZigLLVM_x86)
+                return LLVMX86FastcallCallConv;
+            return LLVMCCallConv;
+        case CallingConventionVectorcall:
+            if (g->zig_target->arch == ZigLLVM_x86)
+                return LLVMX86VectorCallCallConv;
+        // XXX Enable this when the C API exports this enum member too
```

Comment:
We should work around this with `zig_llvm.h`/`zig_llvm.cpp`. The official LLVM C API always lags behind the C++ API.

In fact: 

https://github.com/ziglang/zig/blob/86ba8c06bff84ac4865cef18080fa64fe946cb11/src/zig_llvm.h#L74-L75

https://github.com/ziglang/zig/blob/86ba8c06bff84ac4865cef18080fa64fe946cb11/src/zig_llvm.cpp#L275-L297

Looks like all we're missing is a copy-paste of LLVM's calling convention enum, with some `static_assert`s in zig_llvm.cpp to make sure the values don't change. See src/zig_clang.cpp for inspiration.

Reply:
I've [sent](https://reviews.llvm.org/D71942) a patch that hopefully will get in by the time LLVM 10 is out. But moving away from the C API is also a good idea.